Extensions/PathfindingBehavior/pathfindingruntimebehavior.ts

```diff
@@ -728,21 +874,42 @@ namespace gdjs {
             (obj.getDrawableY() + obj.getHeight() + this._topBorder) /
               this._cellHeight
           );
-          if (
-            topLeftCellX < xPos &&
+
+          objectsOnCell = topLeftCellX < xPos &&
             xPos < bottomRightCellX &&
             topLeftCellY < yPos &&
-            yPos < bottomRightCellY
-          ) {
-            objectsOnCell = true;
-
-            //The cell is impassable, stop here.
-
-            //Superimpose obstacles
+            yPos < bottomRightCellY;
+          }
+          else {
+            this._aabb.min[0] = nodeCenter[0] - this._leftBorder;
+            this._aabb.min[1] = nodeCenter[1] - this._topBorder;
+            this._aabb.max[0] = nodeCenter[0] + this._rightBorder;
+            this._aabb.max[1] = nodeCenter[1] + this._bottomBorder;
+            this._obstacles.getAllObstaclesAroundAABB(this._aabb, this._closeObstacles);
```

Comment:
Wasn't this done just before? Or I'm getting confused about this is used for? Especially considered this will be called for every `this._closeObstacles` member, and it's writing in `this._closeObstacles`, this looks suspicious?

Reply:
I really can't do a copy/paste correctly :sweat_smile: